zlib version check checks system zlib even if other zlib was requested

Build fails when OS provides old version of zlib (e.g. OS X 10.9 Mavericks), even if new version of zlib is provided elsewhere (e.g. by MacPorts).

## Expected Behavior
Successful build

## Current Behavior
Build fails:
```
-- Found ZLIB: /opt/local/lib/libz.dylib (found version "1.2.11") 
-- Performing Test ZLIB_SUPPORTS_GZIP
-- Performing Test ZLIB_SUPPORTS_GZIP - Failed
CMake Error at CMakeLists.txt:238 (message):
  zlib 1.2.x or higher is required to use the gzip format
```
From CMakeOutput.log:
```
Performing C++ SOURCE FILE Test ZLIB_SUPPORTS_GZIP succeeded with the following output:
Change Dir: /opt/local/var/macports/build/_Users_rschmidt_macports_macports-ports-svn-trunk_security_KeePassXC/KeePassXC/work/build/CMakeFiles/CMakeTmp

Run Build Command:"/opt/local/bin/gmake" "cmTC_0078e/fast"
/opt/local/bin/gmake -f CMakeFiles/cmTC_0078e.dir/build.make CMakeFiles/cmTC_0078e.dir/build
gmake[1]: Entering directory '/opt/local/var/macports/build/_Users_rschmidt_macports_macports-ports-svn-trunk_security_KeePassXC/KeePassXC/work/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_0078e.dir/src.cxx.o
/usr/bin/clang++    -pipe -Os -stdlib=libc++ -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -f
visibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-s
ecurity -std=c++11 -stdlib=libc++ -DZLIB_SUPPORTS_GZIP -arch x86_64 -mmacosx-version-min=10.12   -o CMakeFiles/cmTC_0078e.dir/src.cxx.o -c /opt/local/var/macports/build/_
Users_rschmidt_macports_macports-ports-svn-trunk_security_KeePassXC/KeePassXC/work/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_0078e
/opt/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_0078e.dir/link.txt --verbose=1
/usr/bin/clang++  -pipe -Os -stdlib=libc++ -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvi
sibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -D_FORTIFY_SOURCE=2 -Werror=format-sec
urity -std=c++11 -stdlib=libc++ -DZLIB_SUPPORTS_GZIP -arch x86_64 -mmacosx-version-min=10.12 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl
,-headerpad_max_install_names  CMakeFiles/cmTC_0078e.dir/src.cxx.o  -o cmTC_0078e 
gmake[1]: Leaving directory '/opt/local/var/macports/build/_Users_rschmidt_macports_macports-ports-svn-trunk_security_KeePassXC/KeePassXC/work/build/CMakeFiles/CMakeTmp'

Source file was:

  #include <zlib.h>

  #if !defined(ZLIB_VERNUM) || (ZLIB_VERNUM < 0x1200)
      #error zlib 1.2.x or higher is required to use the gzip format
  #endif

  int main() { return 0; }
```

## Possible Solution
The check for zlib version does not use the `CPPFLAGS` that we have provided to cmake, so it doesn't know that it should look for zlib.h in a different directory (/opt/local/include, not /usr/include). (It does, however, use our `LDFLAGS`.)

## Steps to Reproduce (for bugs)
1. Install MacPorts, Xcode and the command line tools on OS X Mavericks
2. Run `sudo port install KeePassXC`

## Context
https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/33014

## Debug Info
<!--- Paste debug info from Help → About here -->
KeePassXC - VERSION
Revision: REVISION

Libraries:
- LIBS

Operating system: OS
CPU architecture: ARCH
Kernel: KERNEL

Enabled extensions:
- EXTENSIONS

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,6 +225,8 @@ find_package(Gcrypt 1.6.0 REQUIRED)
 
 find_package(ZLIB REQUIRED)
 
+set(CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIR})
+
 check_cxx_source_compiles("
   #include <zlib.h>
 

